iotedgehubdev/utils.py

```diff
@@ -22,27 +22,58 @@
 
 class Utils(object):
     @staticmethod
-    def parse_device_connection_str(connection_string):
-        parts = connection_string.split(';')
-        if len(parts) > 0:
-            data = dict()
-            for part in parts:
-                if "=" in part:
-                    subparts = [s.strip() for s in part.split("=", 1)]
-                    data[subparts[0]] = subparts[1]
+    def parse_connection_strs(device_conn_str, hub_conn_str=None):
+        data = Utils._parse_device_connection_str(device_conn_str)
+        data[EC.DEVICE_ACCESS_KEY_KEY] = data.pop(EC.ACCESS_KEY_KEY)
+        if hub_conn_str is not None:
+            hub_data = Utils._parse_hub_connection_str(hub_conn_str, data[EC.HOSTNAME_KEY])
+            data[EC.HUB_ACCESS_KEY_KEY] = hub_data[EC.ACCESS_KEY_KEY]
+            data[EC.ACCESS_KEY_NAME] = hub_data[EC.ACCESS_KEY_NAME]
+        return data
 
+    @staticmethod
+    def _parse_device_connection_str(connection_string):
+        data = Utils._split_connection_string(connection_string)
+        if len(data) > 0:
             if EC.HOSTNAME_KEY not in data or EC.DEVICE_ID_KEY not in data or EC.ACCESS_KEY_KEY not in data:
-                if "SharedAccessKeyName" in data:
+                if EC.ACCESS_KEY_NAME in data:
                     raise KeyError('Please make sure you are using a device connection string '
                                    'instead of an IoT Hub connection string')
                 else:
                     raise KeyError('Error parsing connection string. '
                                    'Please make sure you wrap the connection string with double quotes when supplying it via CLI')
-
             return data
         else:
             raise KeyError('Error parsing connection string')
 
+    @staticmethod
+    def _parse_hub_connection_str(hub_connection_string, host_name):
+        hub_data = Utils._split_connection_string(hub_connection_string)
+        if len(hub_data) > 0:
+            if EC.HOSTNAME_KEY not in hub_data or EC.ACCESS_KEY_NAME not in hub_data or EC.ACCESS_KEY_KEY not in hub_data:
+                if EC.DEVICE_ID_KEY in hub_data:
+                    raise KeyError('Please make sure you are using a IoT Hub connection string '
+                                   'instead of an device connection string')
+                else:
+                    raise KeyError('Error parsing connection string. '
+                                   'Please make sure you wrap the connection string with double quotes when supplying it via CLI')
+            elif hub_data[EC.HOSTNAME_KEY] != host_name:
+                raise KeyError('Please make sure the device belongs to the IoT Hub')
```

Comment:
Is it better to move host_name check to `parse_connection_strs` function? So `_parse_hub_connection_str` would be a pure parse function and can be reused by other functions which doesn't require checking host_name.

Reply:
Thanks for suggestion. The current logic is put the all the check logic in the private method. So the public api just pick the result from the private api. And in iotedgehubdev tool, hub string needs has a strong dependency on the device string. So i would prefer to put the check here, to make sure it is a valid hub string. Besides, the common parse connection logic is already extracted to _split_connection_string method